setup/main.go

```diff
@@ -0,0 +1,1653 @@
+		if verbose {
+			logger.VerboseOutput = outputBoth
+		} else {
+			logger.VerboseOutput = log
+		}
+	} else {
+		logger = NewLogger()
+		if verbose {
+			logger.VerboseOutput = os.Stdout
+		}
+	}
+	setupState.Logger = &logger
+
+	// TODO: check for root?
+
+	setupState.Log(`Welcome to the pganalyze collector installer!
+
+We will go through a series of steps to set up the collector to monitor your
+Postgres database. We will not make any changes to your database or system
+without confirmation.
+
+At a high level, we will:
+
+ 1. Configure a database user and helper functions for the collector, with minimal access
+ 2. Update the collector configuration file with these settings
+ 3. Set up the pg_stat_statements extension in your database for basic query performance monitoring
+ 4. (Optional) Make log-related configuration settings changes to enable our Log Insights feature
+ 5. (Optional) Set up EXPLAIN plan collection to enable our Automated EXPLAIN feature
+
+At each step, we'll check if any changes are necessary, and if so, prompt you to
+provide input or confirm any required changes.
+
+Changes to Postgres configuration settings will be done with the ALTER SYSTEM command.
+If you later need to refine any of these, make sure to use ALTER SYSTEM or ALTER SYSTEM RESET,
+since otherwise, the ALTER SYSTEM changes will override any direct config file edits. Learn
+more at https://www.postgresql.org/docs/current/sql-altersystem.html .
+
+You can stop at any time by pressing Ctrl+C.
+
+If you stop before completing setup, you can resume by running the installer
+again. We can pick up where you left off.`)
+	setupState.Log("")
+	var doSetup bool
+	err := survey.AskOne(&survey.Confirm{
+		Message: "Continue with setup?",
+		Default: false,
+	}, &doSetup)
+	if err != nil {
+		setupState.Log("  automated setup failed: %s", err)
+	}
+	if !doSetup {
+		setupState.Log("Exiting...")
+		os.Exit(0)
+	}
+
+	for _, step := range steps {
+		err := doStep(&setupState, step)
+		if err != nil {
+			os.Exit(1)
+		}
+	}
+}
+
+func doStep(setupState *SetupState, step *Step) error {
+	if step.Check == nil {
+		panic("step missing completion check")
+	}
+	setupState.Logger.StartStep(step.Description)
+	defer setupState.Logger.EndStep()
+	done, err := step.Check(setupState)
+	if err != nil {
+		setupState.Log("✗ failed to check status: %s", err)
+		return err
+	}
+	if done {
+		setupState.Verbose("✓ no changes needed")
+		return nil
+	}
+	if step.Run == nil {
+		// panic because we should always define a Run func if a check can fail
+		panic("check failed and no resolution defined")
+	}
+	setupState.Verbose("? suggesting resolution")
+
+	err = step.Run(setupState)
+	if err != nil {
+		return err
+	}
+
+	setupState.Verbose("  re-checking...")
+	done, err = step.Check(setupState)
+	if err != nil {
+		setupState.Log("✗ failed to check status: %s", err)
+		return err
+	}
+	if !done {
+		err := errors.New("check still failed after running resolution; please try again")
```

Comment:
Its not clear to me how one should respond when this happens - I did everything right (I think?) and I keep getting this:

```
? Continue with setup? Yes
? Select Postgres superuser to connect as for configuration purposes postgres
? Proceed to configuring optional Log Insights feature? Yes
? Proceed to configuring optional Automated EXPLAIN feature? Yes
? The collector configuration must be reloaded for changes to take effect; reload now? Yes
? Review auto_explain configuration settings? Yes
? Set auto_explain.log_analyze to (will be saved to Postgres): on (recommended)
? Set auto_explain.log_buffers to (will be saved to Postgres): on (recommended)
? Set auto_explain.log_timing to (will be saved to Postgres): on
? Set auto_explain.log_triggers to (will be saved to Postgres): on (recommended)
? Set auto_explain.log_verbose to (will be saved to Postgres): on (recommended)
? Set auto_explain.log_format to (will be saved to Postgres): text
? Set auto_explain.log_min_duration, in milliseconds, to (will be saved to Postgres): 
? Set auto_explain.log_nested_statements to (will be saved to Postgres): on (recommended)
  ✗ check still failed after running resolution; please try again
```

Can we show a more verbose error here?

Reply:
So, as discussed, there's a bug here. I'll fix that, but more generally, in terms of giving more info for why re-checks fail, I'm not sure there's a good way to add meaningful, actionable feedback here. There's no error per se: we run the check for the step, if that does not pass, we run the resolution (typically requiring user input) and then run the check again. If the check still does not pass, the reason should be obvious from the nature of the check (if not, we may have made the step too big). (If there is an actual error running the check--e.g., we can't connect to the database--we do print that.)

Let's review again once this is working as intended and then figure out how to address this (maybe we should prompt you to contact support in these situations, or re-run with the --verbose or --log flag and then contact support?).